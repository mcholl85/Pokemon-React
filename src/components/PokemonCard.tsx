import { Link } from 'react-router-dom'
import { SPRITE_EXTENSION, SPRITE_URL } from '../constants'
import PokemonImg from './PokemonImg'

type PokemonCardProps = {
  order: number
  name: string
}

function PokemonCard({ order, name }: PokemonCardProps) {
  const imgSrc = `${SPRITE_URL}${order}${SPRITE_EXTENSION}`

  return (
    <div key={order} className='col-xl-3 col-lg-4 col-sm-6 pb-md-5 pb-3'>
      <Link to={`/${name}`} className='text-decoration-none'>
        <div className='card pokeball-card rounded-3 border-0 overflow-hidden shadow grayscale-background'>
          <header className='d-flex align-items-center justify-content-end pt-2 px-3 grayscale-medium'>
            #{order}
          </header>
          <div className='p-1 p-sm-3 rounded-3 d-flex align-items-center justify-content-center'>
            <PokemonImg imgSrc={imgSrc} height='150' width='150' alt='pokemon image' />
          </div>
          <h4 className='pt-1 pt-sm-3 mb-4 text-center grayscale-dark'>{name}</h4>
        </div>
      </Link>
    </div>
  )
}
export default PokemonCard
