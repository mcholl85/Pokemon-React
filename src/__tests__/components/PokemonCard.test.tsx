import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import PokemonCard from '../../components/PokemonCard'

describe('PokemonCard', () => {
  test('Matches DOM snapshots', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PokemonCard order={2} name={'test'} imgSrc={''} />
      </MemoryRouter>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test.each`
    order | name
    ${1}  | ${'test1'}
    ${2}  | ${'test2'}
    ${3}  | ${'test3'}
  `('Should render the pokemon order: $order & name: $name values', ({ order, name }) => {
    render(
      <MemoryRouter>
        <PokemonCard order={order} name={name} imgSrc={''} />
      </MemoryRouter>,
    )

    expect(screen.getByRole('heading', { name: name })).toHaveTextContent(name)
    expect(screen.getByRole('banner')).toHaveTextContent(`#${order}`)
  })
})
