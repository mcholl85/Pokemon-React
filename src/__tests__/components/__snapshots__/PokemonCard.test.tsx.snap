// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard Matches DOM snapshots 1`] = `
<DocumentFragment>
  <div
    class="col-xl-3 col-lg-4 col-sm-6 pb-md-5 pb-3"
  >
    <a
      class="text-decoration-none"
      href="/test"
    >
      <div
        class="card pokeball-card rounded-3 border-0 overflow-hidden shadow grayscale-background"
      >
        <header
          class="d-flex align-items-center justify-content-end pt-2 px-3 grayscale-medium"
        >
          #2
        </header>
        <div
          class="p-1 p-sm-3 rounded-3 d-flex align-items-center justify-content-center"
        >
          <img
            alt="pokemon image"
            height="150"
            src=""
            width="150"
          />
        </div>
        <h4
          class="pt-1 pt-sm-3 mb-4 text-center grayscale-dark"
        >
          test
        </h4>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
